Gems/ImGui/Code/Source/ImGuiManager.cpp

```diff
@@ -729,10 +715,19 @@ void ImGuiManager::ToggleThroughImGuiVisibleState()
         case DisplayState::Visible:
             m_clientMenuBarState = DisplayState::Hidden;
 
-            // Restore old cursor state
-            AzFramework::InputSystemCursorRequestBus::Event(AzFramework::InputDeviceMouse::Id,
-                &AzFramework::InputSystemCursorRequests::SetSystemCursorState,
-                m_previousSystemCursorState);
+            const bool inGame = !gEnv->IsEditor() || gEnv->IsEditorGameMode(); 
+
+            // Restore old cursor state if the "In Game" state hasn't changed.
+            // This prevents incorrectly hiding the cursor when a user is in
+            // game mode, opens the console, exits game mode, and then hides the
+            // console resulting in the cursor being restored to what it was
+            // in game mode, usually making the cursor hidden in the editor.
+            if (m_previousSystemCursorStateInGame == inGame)
+            {
+                AzFramework::InputSystemCursorRequestBus::Event(AzFramework::InputDeviceMouse::Id,
+                    &AzFramework::InputSystemCursorRequests::SetSystemCursorState,
+                    m_previousSystemCursorState);
+            }
```

Comment:
Since this is all for Editor game mode which is used for development, we should bias the behavior for showing a cursor when it shouldn't be shown as that is the safer approach.

Reply:
I almost set this code to always show the cursor in Editor when not in game, but wasn't sure if maybe there's some case where the cursor is hidden and that's why the code was written this way?

Reply:
I can't think of a case where the cursor should remain hidden outside of game mode if in the Editor.

Suggestion: Instead of `m_previousSystemCursorStateInGame`, how about `m_inGameWhenImGuiActivated`? While this logic is only affecting the cursor, the actual logic check here is "did the game mode state change between when we made ImGui visible and when we made it hidden", so a name tied to the logic check instead of the action we want to take might make this a little bit more readable as well.

Reply:
I went ahead and updated this so we just never hide the cursor if we're not in game.